performance-es.md

```diff
@@ -57,3 +57,123 @@ La optimización prematura también puede afectarte al atarte a ciertas decision
+  consulta. Si necesitas dos tipos de consultas, disponer de una "vista"
+  adicional a tus datos puede ser una gran mejora. Por ejemplo, un set de
+  structs puede tener un ID primario (integer) que usas para buscar en un
+  slice, pero a veces necesitas buscar por un ID secundario (string). En lugar
+  de iterar sobre el slice, puedes mejorar tu estructura de datos con un mapa de
+  string a ID o directamente al struct en cuestión.
+
+* Información adicional sobre los elementos
+
+  Por ejemplo, mantener un filtro de Bloom de todos los elementos que has
+  insertado puede permitirte retornar rápidamente "sin coincidencias" a las
+  búsquedas. Estos necesitan ser pequeños y rápidos para no abrumar el resto de
+  la estructura de datos. (Si una búsqueda en tu estructura de datos principal
+  es barata, el costo del filtro de Bloom superará cualquier ahorro.)
+
+* Si las búsquedas son costosas, agrega una cache
+
+  A mayor escala, una cache interna o externa (como memcache) puede ayudar.
+  Puede ser excesivo para una única estructura de datos. Hablaremos más sobre
+  caches más adelante.
+
+Este tipo de cambios son útiles cuando los datos que necesitan son baratos de
+almacenar y fáciles de mantener actualizados.
+
+Estos son todos ejemplos claros de "realizar menos trabajo" a nivel de
+la estructura de datos. Todos cuestan espacio. La mayoría de las veces, si estás
+optimizando para CPU, tu programa usará más memoria. Se trata del clásico [space-time trade-off](https://en.wikipedia.org/wiki/Space%E2%80%93time_tradeoff).
+
+Si tu programa utiliza demasiada memoria, también es posible ir por el otro
+camino. Reduce el uso de espacio a cambio de una mayor carga computacional. En
+lugar de almacenar cosas, calcúlalas cada vez. También puedes comprimir los datos
+en memoria y descomprimirlos cuando los necesites.
+
+[Small Memory Software](http://smallmemory.com/book.html) es un libro disponible
+online que cubre técnicas para reducir el espacio utilizado por tus programas.
+Aunque fue originalmente escrito dirigído a desarrolladores de sistemas
+embebidos, sus ideas son aplicables para programas en hardware moderno que
+manejen gran cantidad de datos.
+
+* Reorganiza tus datos
+
+  Elimina el padding. Remueve campos extra. Utiliza tipos de datos mas pequeños.
+
+* Cambia a una estructura de datos más lenta
+
+  Estructuras de datos más simples frecuentemente presentan menores
+  requerimientos de memoria. Por ejemplo, cambiar una estructura tipo arbol con
+  uso extensivo de punteros a un slice y búsqueda lineal.
+
+* Compresión a medida para tus datos
+
+  Los algoritmos de compresión dependen fuertemente de qué este siendo comprimido. Lo mejor es
+  elegir uno que se ajuste a tus datos. Si tienes un []byte, entonces algo como snappy, gzip, lz4,
+  funciona bien. Para datos de punto flotante existe go-tsz para series temporales y fpc para datos
+  scientificos. Se ha realizado mucha investigación sobre la compresion de integers, generalmente
```

Comment:
```suggestion
  científicos. Se ha realizado mucha investigación sobre la compresión de integers, generalmente
```

Since `integers` here refers to the mathematical term, not to the Go type, it might be better to translate it as `números enteros`